packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts

```diff
@@ -390,8 +390,12 @@ export class AnnotationsStateMachine extends StateMachine<States, AnnotationType
 
                 delete: () => {
                     if (this.active == null) return;
+                    this.transition('cancel');
```

Comment:
This is an anti-pattern, we should not transition inside these events. Partly as that could easily create an infinite loop, but also as it goes against the design of state machines.

Reply:
Fair. Another solution could be to store the `visible: boolean` state in the `TextInput` class and check that instead. I'll do that.

Reply:
Done.